Sources/TimeFreeze.swift

```diff
@@ -20,6 +20,14 @@ struct TimeFreeze {
         return (TimeFreeze.systemUptime() - self.uptime) + self.timestamp
     }
 
+    /// Returns the time plus analytical metadata about the time.
+    ///
+    /// See explanation of fields in the comment for the FullTime struct fore more info.
+    var fullTime: FullTime? {
+        return FullTime(date: Date(timeIntervalSince1970: adjustedTimestamp),
+                        timeSinceLastNtpSync: TimeFreeze.systemUptime() - uptime)
```

Comment:
This is optional: I prefer for `TimeFreeze` to know nothing about composing this, but expose the `timeSinceLastNtpSync` instead. So you'd have `var timeSinceLastNtpSync { ... }` here and compose this `FullTime` thing on the `Clock` interface by doing something like:

```swift
public static var fullTime: FullTime? {
    guard let stableTime = self.stableTime else {
        return nil
    }

    return FullTime(date: Date(timeIntervalSince1970: stableTime.adjustedTimestamp),
                    timeSinceLastNtpSync: TimeFreeze.systemUptime() - stableTime.uptime)
}
```

Reply:
done!